mesonbuild/programs.py

```diff
@@ -123,7 +123,7 @@ def get_version(self, interpreter: T.Optional['Interpreter'] = None) -> str:
             if not output:
                 output = e.strip()
 
-            match = re.search(r'([0-9](\.[0-9]+)+)', output)
+            match = re.search(r'([0-9]+(\.[0-9]+)+)', output)
```

Comment:
I wonder if would be simpler?
```suggestion
            match = re.search(r'([0-9]+\.[0-9\.]+)', output)
```

Reply:
That'd also match things like `1..` though.

Reply:
Right. But the original regex was doing this. Also, if you match versions with dots in first regex, the second regex could simply be `([0-9]+)`...

Reply:
Which after thinking is the same behaviour as before due to the fallback behaviour. But in any case, I personally don't think either of these are significantly easier to read than the other, but am happy to adapt if anyone feels strongly about it.

Reply:
Ah, the problem though is that it would basically reintroduce the issue that proper dotted version numbers wouldn't be favored over weird version numbers as `1..`. So if we had `foo 1.. 1.2.3` we'd return `1..` and not `1.2.3`.

Reply:
ok. I guess it's better to keep your original regex, and to remove the dot from the second one then.